packages/exchange-core/src/MatchingEngine.ts

```diff
@@ -1,31 +1,27 @@
 import { List } from 'immutable';
 import { Subject } from 'rxjs';
-import { IRECDeviceService, LocationService } from '@energyweb/utils-general';
+import { IDeviceService, ILocationService } from '@energyweb/utils-general';
 
 import { OrderSide, Order, OrderStatus } from './Order';
 import { Trade } from './Trade';
 import { Product } from './Product';
 import { Bid } from './Bid';
 import { Ask } from './Ask';
 
-export type Listener<T> = (entity: T) => void;
-
-type ExecutedTrade = { trade: Trade; askKey: number; bidKey: number; isPartial: boolean };
-
-export class Matching {
-    private deviceService = new IRECDeviceService();
-
-    private locationService = new LocationService();
+export type TradeExecutedEvent = { trade: Trade; ask: Ask; bid: Bid };
 
+export class MatchingEngine {
     private bids: List<Bid> = List<Bid>();
 
     private asks: List<Ask> = List<Ask>();
 
-    public trades = new Subject<List<Trade>>();
+    public trades = new Subject<List<TradeExecutedEvent>>();
 
     public cancellationQueue = List<string>();
 
-    public submitOrder(order: Order) {
+    constructor(private deviceService: IDeviceService, private locationService: ILocationService) {}
```

Comment:
If these are private and not changed later, then maybe they should be: `private readonly`?